src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/SolutionExplorer_OutOfProc.cs

```diff
@@ -43,7 +43,10 @@ public void OpenSolution(string path, bool saveExistingSolutionIfExists = false)
             => _inProc.OpenSolution(path, saveExistingSolutionIfExists);
 
         public void AddProject(ProjectUtils.Project projectName, string projectTemplate, string languageName)
-            => _inProc.AddProject(projectName.Name, projectTemplate, languageName);
+        {
+            _inProc.AddProject(projectName.Name, projectTemplate, languageName);
+            _instance.Workspace.WaitForAsyncOperations(FeatureAttribute.Workspace);
```

Comment:
What asynchrony in the workspace is this waiting for?

Reply:
I don't remember exactly.